pkg/types/validation/installconfig.go

```diff
@@ -156,6 +156,21 @@ func ValidateInstallConfig(c *types.InstallConfig, usingAgentMethod bool) field.
 		}
 	}
 
+	if c.Capabilities != nil {
+		if c.Capabilities.BaselineCapabilitySet == configv1.ClusterVersionCapabilitySetNone {
+			capabilityMap := make(map[configv1.ClusterVersionCapability]struct{}, len(c.Capabilities.AdditionalEnabledCapabilities))
+			for _, cap := range c.Capabilities.AdditionalEnabledCapabilities {
+				capabilityMap[cap] = struct{}{}
+			}
```

Comment:
This might be easier to manage with the [sets utils](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets)

```
enabledCaps: sets.New[configv1.ClusterVersionCapability](c.Capabilities.AdditionalEnabledCapabilities...)
if enabledCaps.Has(configv1.ClusterVersionCapabilityBaremetal) && !enabledCaps.Has(configv1.ClusterVersionCapabilityMachineAPI) {
    allErrs = append(allErrs, ...)
}
```